src/main/java/org/synyx/urlaubsverwaltung/absence/DateRange.java

```diff
@@ -25,6 +41,92 @@ public LocalDate getEndDate() {
         return endDate;
     }
 
+    /**
+     * Checks if this date ranges overlap each other.
+     *
+     * @param dateRange to check for overlap
+     * @return true if the date ranges overlap, false otherwise
+     */
+    public boolean isOverlapping(final DateRange dateRange) {
+        return (startDate.isBefore(dateRange.endDate) || startDate.isEqual(dateRange.endDate)) &&
+            (dateRange.startDate.isBefore(endDate) || dateRange.startDate.isEqual(endDate));
+    }
+
+    /**
+     * Returns the overlapping date range of the date ranges.
+     *
+     * <p>
+     * Date ranges are inclusive of the start and the end date.
+     * A date range overlaps another if it shares some common part of the
+     * dates. This method returns the amount of the overlap,
+     * only if the date ranges actually do overlap.
+     * If the date ranges do not overlap, a empty {@link DateRange} is returned.
+     * <p>
+     * When two date ranges are compared the result is one of three states:
+     * (a) they abut
+     * (b) there is a gap between them
+     * (c) they overlap.
+     * The abuts state takes precedence over the other two, thus a zero duration
+     * date range at the start of a larger date range abuts and does not overlap.
+     *
+     * @param dateRange overlapping date range
+     * @return the overlapping date rage
+     */
+    public DateRange overlap(final DateRange dateRange) {
+        if (!isOverlapping(dateRange)) {
+            return new DateRange(null, null);
```

Comment:
Whats the advantage over `Optional<DateRage>` as return type?

Reply:
good point. I will take a look at it